language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/semantictokens/SemanticTokensVisitor.java

```diff
@@ -431,14 +439,12 @@ private boolean isReadonly(Node node) {
      * @param startLine Start line position
      */
     private void processSymbols(Node node, LinePosition startLine) {
-        if (semanticTokens.contains(new SemanticToken(startLine.line(), startLine.offset()))) {
-            return;
-        }
-        Optional<SemanticModel> semanticModel = this.semanticTokensContext.currentSemanticModel();
-        if (semanticModel.isEmpty()) {
+        if (semanticTokens.contains(new SemanticToken(startLine.line(), startLine.offset()))
```

Comment:
> Can't we do this check without creating new objects for `SemanticToken`? It is better if we can do this since we create so many objects even though they are GCed

We are using the SemanticToken object's equal() method to compare the quality between objects within the SemanticTokens TreeSet. Hence, we need an object. If we are to check the contains() separately, we have to implement the tree traversal by ourselves, which will not be as fast as the java implementation.